### PR TITLE
feature: add new filter to bulding search

### PR DESCRIPTION
**Pull request by kamilfedio**
#97 
I added the filters mentioned and improve graphql&model of bulidings

### CODE REVIEW

File: lib/features/buildings_map/controllers.dart [hunk 1]
```diff
@@ -34,7 +34,11 @@ class BuildingsListViewController extends _$BuildingsListViewController
   @override
   bool filterMethod(BuildingModel item, String filterStr) {
     return item.name.toLowerCase().contains(filterStr) ||
-        (item.addres?.toLowerCase().contains(filterStr) ?? false);
+        (item.addres?.toLowerCase().contains(filterStr) ?? false) ||
+        (item.naturalName?.toLowerCase().contains(filterStr) ?? false) ||
+        (!filterStr.contains('-')
+            ? item.name.toLowerCase().replaceAll('-', '').contains(filterStr)
```
Comment (simon-the-shark) [inline lib/features/buildings_map/controllers.dart:40]: it stopped being easily readable at this point and I would strongly recomend some refactor to divide this into smaller logically separated pieces.

so you can do this in many ways, but my personal proposition would be:

in map view config file
```dart
abstract class BuildingSearchConfig {
  static const buildingCodeSeparator = "-";
}
```

then create util file, example location: lib\features\buildings_map\utils.dart
```dart
extension ContainsCaseUnsensitive on String? {
  bool containsUnsensitive(String str) {
    if (this == null) return false;
    return this!.toLowerCase().contains(str.toLowerCase());
  }

bool containsBuildingCode(String buildingCode) {
    if (buildingCode.contains(BuildingSearchConfig.buildingCodeSeparator)) {
      return containsUnsensitive(buildingCode);
    }
    return this
            ?.replaceAll(BuildingSearchConfig.buildingCodeSeparator, "")
            .containsUnsensitive(buildingCode) ??
        false;
  }
}
```
and the usage would be:
```dart
  @override
  bool filterMethod(BuildingModel item, String filterStr) {
    return item.name.containsBuildingCode(filterStr) ||
        item.addres.containsUnsensitive(filterStr) ||
        item.naturalName.containsUnsensitive(filterStr);
  }
```

it's just a proposition, you can refactor it in some other way you find suitable. Also I hope I haven't broken the logic :)

**Review by simon-the-shark — COMMENTED** [review]
good work. I have one proposition of a simple refactor.

**Comment by kamilfedio** [comment]
@simon-the-shark Okay, I used your idea

**Comment by simon-the-shark** [comment]
You can merge :)

**Pull request merged**